app.py

```diff
@@ -231,9 +233,26 @@ def before_request_https():
             # We use a 302 in case we need to revert the redirect.
             return redirect(url, code=302)
 
+def version():
+    """Get the version from the Heroku environment variables."""
+    if not os.getenv('DYNO'):
+        # Not on Heroku
+        return 'DEV'
+
+    vrz = os.getenv('HEROKU_RELEASE_CREATED_AT')
+    the_date = datetime.date.fromisoformat(vrz[:10]) if vrz else datetime.date.today()
+
+    commit = os.getenv('HEROKU_SLUG_COMMIT', '????')[0:6]
+    return the_date.strftime('%b %d') + f' ({commit})'
+
 # Unique random key for sessions.
 # For settings with multiple workers, an environment variable is required, otherwise cookies will be constantly removed and re-set by different workers.
-app.config['SECRET_KEY'] = os.getenv ('SECRET_KEY') or uuid.uuid4().hex
+if version() == 'DEV':
+    app.config['SECRET_KEY'] = os.getenv ('SECRET_KEY') or uuid.uuid4().hex
```

Comment:
`os.getenv()` takes a second argument, which is what should be returned if there is no environment variable with the given name. So this could simplify (a little) to:

```suggestion
    app.config['SECRET_KEY'] = os.getenv('SECRET_KEY', uuid.uuid4().hex)
```